### DISCUSSION TITLE
Importing products

### DISCUSSION THREAD

**Discussion by Rodien** [openstore-ecommerce/OpenStore#141] [General]
Has anyone tried importing 2000+ products into openstore?

I keep getting a timeout error. 
I bump up the connection timeout in the connection string and the import improve a little bit but I still get the timeout error. 

And if I import like 800 products, they do not show right away in the system.  They show up like after 10 min or so.

Does openstore have a limit for importing products?

**Comment by leedavi** [comment]
I've imported 8000+  and it worked OK... but took a LONG LONG time!!!

You need to up all the timeouts in the web.config.  I made them BIG and then lowered them after the import.

Look in you web.config for 

executionTimeout="xxxxx"
maxRequestLength="xxxxxxxx"


There is also a report of products not going into the correct category.  I have seen it on a site with about 1000+ products, that I imported from v3. So, keep an eye on that.  I think the problem I had was related to data being corrupt, but I'm unsure.

**Reply by Rodien** [reply]
I will give it a try

**Reply by Rodien** [reply]
This did not work for me. 

The import process keeps timing out at 1 min and 30 seconds.

I have done a few tests and it seems that the import process is active even though it had just thrown a timeout. 

The import takes a day or so to complete. And sometimes nothing gets imported. 

I am running SQL server Profiler to try and understand what is going on.

**Reply by leedavi** [reply]
"A Day"...... wow!!!...NEVER seen it be that long.  Are you doing the image zip at the same time?

You can update the images by simply unzipping on the server. and do not include on the import.  

This could be an infinite loop on categories... child point to a parent and the child.   Even when I did 8000+ products I think it only took 2-3 hours.  800 products should not be longer than that, unless you server is running out of memory.

**Reply by Rodien** [reply]
I forget to mention that I am using a custom import plugin. ** You made it for us** many thanks for that. 

After 1 min and 30 secs it throws the following error in the logbook: The transaction log for database 'openstore-dev' is full due to 'CHECKPOINT'. 

I removed the categories code and I did manage to import 575 products before the timeout. (after 1 min 30 sec)

I am trying to figure out this error (The transaction log for database 'openstore-dev' is full due to 'CHECKPOINT') at the moment

**Reply by leedavi** [reply]
OK, was the import module made for NBS ?  Is it on GitHub ?

**Reply by Rodien** [reply]
It's a NBS plugin. 

I send you an invite to the repository.

**Reply by leedavi** [reply]
I've not had any invite?

**Reply by Rodien** [reply]
Hmm weird. 

The status is "Pending Invite";

![image](https://user-images.githubusercontent.com/50739829/114561947-574e0a80-9c6e-11eb-9aaf-06dcf4d83d8d.png)

**Reply by Rodien** [reply]
[Repository](https://github.com/Bond-for-web-solutions/OpenStoreCSVtransfer)

**Reply by leedavi** [reply]
OK, I got that invite.  I remember this one.  CSV import,  I thought you meant the XML one.

There was some trouble with csvhelper.dll being an old version is your install.   I sent an email about it.  I think I completed the work by using the old version you were using.  Maybe with an upgrade this clash has broken things.  Or obviously something else.

This was written over 2 years ago.  Has it been working up to the point you upgraded something?  Or did it just stop ?

Contact me by email and we can try to work on a solution.

**Reply by Rodien** [reply]
I send you an email.

**Reply by ufoloko** [reply]
Hi Rodien, can yo please share again the import plugin, should be great. Thanks

**Reply by Rodien** [reply]
I'll talk to @leedavi to see how we can make the code available on the Openstore GitHub account > Hi Rodien, can yo please share again the import plugin, should be great. Thanks

**Comment by Rodien** [comment]
Hey David, 

One more question😁

I am trying to make a scheduler for the import. This way I can avoid the httpcontext...

But "GategoryUtils.GetCategoryData" is throwing an Object reference not set to an instance of an object.'

Am guessing that it is throwing this error because I am not using http context anymore. 

![image](https://user-images.githubusercontent.com/50739829/114642866-e857cc80-9cd4-11eb-8acb-770212b66b56.png)

`// overload to support out of http context instantiations ie. when the scheduler runs NBrightDnnIdx
        public ProductData(int productId, int portalId, String lang, Boolean hydrateLists = true, String typeCode = "PRD")
        {
            _portalId = portalId;
            ResetData(productId, lang, hydrateLists, typeCode);
        }

`

Does OpenStore has something similar as the above code but for the "GetCategoryData" method?

**Reply by leedavi** [reply]
It could be the cacheing.

Try 

catData = new CategoryData(categoryId, lang);

And see if that works.,

**Reply by Rodien** [reply]
Ahh that did not work😌
Getting the same error "Object reference not set to an instance of an object."

I tried clearing the cache right before " var categoryData = new CategoryData(-1, culture); " but it did not have any effect.

I had this same error for productData = new ProductData(productid, culture); but supplying PortalID
ProductData(productid, PortalID, culture) fixed it for me

**Comment by leedavi** [comment]
I see.  The scheduler does not have any portals.

You need to loop through all the portals in the install or pass the specific portalid.

**Comment by DNNMonster** [comment]
If you are able to upgrade, you may wish to re-test this with the recent changes to the Product save function.  I think you'll find it's improved.

Don't expect warp speed but a noticed improvement should be there.
